app/models/manageiq/providers/vmware/infra_manager/refresh_parser/filter.rb

```diff
@@ -88,8 +88,9 @@ def vm_inv_by_vm(vm)
     # Since a Folder and a Datacenter are both an EmsFolder
     # we need to handle @vc_data[:folder] and @vc_data[:dc]
     def folder_inv_by_folder(folder)
-      mor = folder.ems_ref_obj
-      return nil if mor.nil?
+      return nil if folder&.ems_ref.nil?
+
+      mor = VimString.new(folder.ems_ref, folder.ems_ref_type, :ManagedObjectReference)
```

Comment:
Can this code stay as it was since you implemented ems_ref_obj?

Reply:
Haven't subclassed EmsFolders yet, once we have [this](https://github.com/ManageIQ/manageiq-providers-vmware/pull/481/files#diff-652aa8a739dfff9d9a8e3b3a59c826a6R1) then yes it will be able to just use `.ems_ref_obj`

Reply:
Fixed by [this](https://github.com/ManageIQ/manageiq-providers-vmware/pull/481/files#diff-aed8ccaa99c322460352295469765b27R91)